### DISCUSSION TITLE
use ohmyposh in bash - WSL 2

### DISCUSSION THREAD

**Discussion by thomasdoerr** [JanDeDobbeleer/oh-my-posh#923] [General]
Good morning,

I just tried out to enable ohmyposh in my linux ubuntu bash, which is running in WSL2 under Windows 10.

This is not working on my machine:
eval "$(oh-my-posh --init --shell bash --config ~/jandedobbeleer.omp.json)"

The path is set correctly, but it must be:
eval "$(oh-my-posh-wsl --init --shell bash --config ~/jandedobbeleer.omp.json)"

The -wsl is missing.

Best
Thomas

**Comment by JanDeDobbeleer** [comment]
@thomasdoerr thx, I added that context to the docs (will be there soon)

**Reply by thomasdoerr** [reply]
Jan, I really like oh-my-posh and used the theme fish in version 2. Since I updated to V3, the git integration in that theme seems to be missing. Not a big thing, but it motivated me to write a theme that fit's my needs.
Could I share that via a Pull Request so that it will be part of the official set of themes? I'm still working on it, just wanted to know, if this is possible?

**Reply by JanDeDobbeleer** [reply]
That's definitely possible! We welcome all additions (if sensible 😅).

**Reply by thomasdoerr** [reply]
super, we'll see if you find it sensible :-) I keep you posted

**Reply by JanDeDobbeleer** [reply]
themes are always OK ;-)

**Reply by Hxx-xxH** [reply]
I've met a issue in wsl2 zsh.Theme jandedobbeleer did not work well.
The errors are:"(eval):1: command not found: Oh   (eval):2: command not found: open".
But it works under root environment. And I've changed other several themes.They all work as I wish.
So,I thought this issue only happens when I use this theme.(Also there may be someting wrong with my settings).
Sorry for my poor English.I hope that I have describe this issue accurately.
Btw,I really like oh-my-posh and this theme,thank you.
![omp](https://user-images.githubusercontent.com/31878519/132515451-c76b2b5f-af12-49cf-958a-6d793e7c8050.png)

**Reply by JanDeDobbeleer** [reply]
This is probably due to the theme being in a location your regular WSL user doesn't have access to.

**Reply by thomasdoerr** [reply]
Can you post the content of your .zshrc? I mean the line in which you enable ohmyposh.

**Reply by thomasdoerr** [reply]
On my machine, this works in .bash_profile, with a theme copied to the linux local home:
eval "$(oh-my-posh-wsl --init --shell bash --config ~/.rudolfs-light.omp.json)"

or in /etc/bash.bashrc
eval "$(/mnt/c/Users/<username>/AppData/Local/Programs/oh-my-posh/bin/oh-my-posh-wsl --init --shell bash --config /home/<username>/rudolfs-light.omp.json)"

But the nerd fonts only work correctly when starting the bash via Windows Terminal.

**Reply by thomasdoerr** [reply]
On macOS with zshrc:
eval "$(oh-my-posh --init --shell zsh --config /usr/local/opt/oh-my-posh/themes/rudolfs-dark.omp.json)"

**Reply by Hxx-xxH** [reply]
I put all the themes in the same folder `/home/hsj(current user)` with `.zshrc` file.
And all of them work well except the theme `jandedobbeleer`.

The content of my .zshrc is:
`export ZSH="/home/hsj/.oh-my-zsh"`
`ZSH_THEME="agnoster"`
`eval "$(oh-my-posh --init --shell zsh --config ~/M365Princess.omp.json)"`

Thanks for your replies.

**Reply by Hxx-xxH** [reply]
I've only installed nerd fonts in Windows,and set nerd font both in wsl and Windows terminal,and both work well.

**Comment by thomasdoerr** [comment]
A difference I see is that you use oh-my-posh which is an exe in the bin folder and I’m using oh-my-posh-wsl which is if I understand correctly a Linux file.



On 9. Sep 2021, at 14:37, Hxx ***@***.***> wrote:

﻿

I put all the themes in the same folder /home/hsj(current user) with .zshrc file.
And all of them work well except the theme jandedobbeleer.

The content of my .zshrc is:
export ZSH="/home/hsj/.oh-my-zsh"
ZSH_THEME="agnoster"
eval "$(oh-my-posh --init --shell zsh --config ~/M365Princess.omp.json)"

Thanks for your replies.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/JanDeDobbeleer/oh-my-posh/discussions/923#discussioncomment-1301303>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABUZMBK3BWYJXJX6EX6DFNDUBCTBHANCNFSM5DFTGJHQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by Hxx-xxH** [reply]
I've tried to use `oh-my-posh-wsl` instead of `oh-my-posh`.
`eval "$(oh-my-posh-wsl --init --shell zsh --config ~/jandedobbeleer.omp.json)"`
But it still happened.
![image](https://user-images.githubusercontent.com/31878519/132717975-928c313f-1c86-495a-b12b-a2f250514414.png)

**Reply by JanDeDobbeleer** [reply]
Please update as I noticed the error format isn't correct when used in eval mode. That should at least display the info correctly.

**Reply by Hxx-xxH** [reply]
Thanks a lot.
I've updated oh-my-posh to the latest version.And there are no more errors raised.
